joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/JavaSrc2Cpg.scala

```diff
@@ -59,25 +104,35 @@ class JavaSrc2Cpg extends X2CpgFrontend[Config] {
       dir.pathAsString
     }
 
-    val delombokSourcesDir = Option.when(config.delombokFullAnalysis || config.delombokTypesOnly) {
+    val delombokSourcesDir = Option.when(runDelombok) {
       Delombok.run(originalSourcesDir, config.delombokJavaHome)
     }
 
     val analysisSourceFilePath =
-      if (config.delombokFullAnalysis)
+      if (runDelombok && (delombokMode != DelombokMode.TypesOnly))
         delombokSourcesDir.get
       else
         originalSourcesDir
 
     val typeSourcesPath =
-      if (config.delombokFullAnalysis || config.delombokTypesOnly)
+      if (runDelombok)
         delombokSourcesDir.get
```

Comment:
we could collapse this whole if into `delombokSourcesDir.getOrElse(originalSourcesDir)`

Reply:
This is true (and avoids the `get` call, which is better)